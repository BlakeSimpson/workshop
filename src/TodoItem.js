import React, { Component } from 'react';

/**
 * This component should render a todo item as an <input type="checkbox" /> with a label
 */
class TodoItem extends Component {
  render() {
    return (
      <div>TodoItem</div>
    );
  }
}

export default TodoItem;
