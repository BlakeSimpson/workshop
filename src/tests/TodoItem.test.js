import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TodoItem from '../TodoItem';

const item = {
  text: 'Get to Las Palmas ðŸŒ´',
  checked: true,
};

it('should render a todo item as a checkbox with a label', () => {
	const tree = toJson(shallow(<TodoItem text={item.text} checked={item.checked} />));
	expect(tree).toMatchSnapshot();
});
