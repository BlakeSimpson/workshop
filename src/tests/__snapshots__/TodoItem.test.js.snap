exports[`test should render a todo item as a checkbox with a label 1`] = `
<label>
  <input
    checked={true}
    onClick={[Function]}
    type="checkbox" />
  Get to the workshop 🚶‍♀️
</label>
`;
